src/time.rs

```diff
@@ -0,0 +1,182 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use std::{ops, fmt};
+
+static NANOS_PER_SEC: u32 = 1_000_000_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TimeStamp {
+    secs: u64,
+    nanos: u32,
```

Comment:
Please let's not hassle with this. `pub struct Timestamp { millis: u64 }` is *more than enough*, we are using it right now for a reason. Nanoseconds are completely unnecessary in a TCP/IP stack and calculations with them are a waste of time, and the additional API complexity is a nightmare.